apps/desks.py

```diff
@@ -321,6 +322,36 @@ def get_desk_name(self, desk_id):
 
         return desk_name
 
+    def get(self, req, lookup):
+        desks = list(super().get(req, lookup))
+
+        members_list = []
+        db_users = app.data.mongo.pymongo('users').db['users']
+
+        # find display_name from the users document for each member in desks document
+        for desk in desks:
+            if 'members' in desk:
+                users = list(db_users.find({
+                    '_id': {'$in': [member['user'] for member in desk['members']]}},
+                    {'display_name': 1}
+                ))
+
+                for user in users:
+                    if not any(item == user for item in members_list):
```

Comment:
I don't think you will get any duplicates from mongo, _id should be unique

Reply:
Actually, two desks can have the same members in that case the user data get duplicate entry in members_list, that's why I have that check for duplicate data.

Reply:
I see, not sure why not handle it per desk then, is it to avoid extra sorting?

Reply:
yeah it's to avoid extra sorting.